File: 11INFORMATIKA/ANDROID/11KOM2/01/BankAccount.java
public class BankAccount {
    
    public static double OverdrawnPenalty = 20.00;
    private String accNumber  = String.format("AC%d", (int) Math.random()*900 + 100);
    private String pinSecurity;
    private double balance = 0;

    public BankAccount(){
        this.pinSecurity = "";
    }

    public BankAccount(String pin){
        this.pinSecurity = pin;
    }

    public double getBalance(){
        return this.balance;
    }

    public void deposit(String pin,double amount){
        if (pin == this.pinSecurity){
            this.balance += amount;
        }
    }

    public void withdraw(){

    }

    public static void changeOverdrwanPenalty(){

    }

    public static void main(String[] args) {
        BankAccount bankAcc1 = new BankAccount();
        BankAccount bankAcc2 = new BankAccount("12345");

        System.out.println(bankAcc1.getBalance());
    }



}

File: 11INFORMATIKA/ANDROID/11KOM1/01/bank_account.py
from random import randint

class BankAccount():

    OVERDRAWN_PENALTY = 20.00

    def __init__(self, pin_security=None):
        self.acc_number = "AC"+str(randint(100, 1000))
        self.pin_security = str(pin_security)
        self.balance = 0

    def get_balance(self):
        return self.balance
    
    def deposit(self, pin, amount):
        if pin == self.pin_security:
            self.balance += amount
        
    def withdraw(self, pin, amount):
        if pin == self.pin_security:
            if self.balance >= amount:
                self.balance -= amount
            else:
                self.balance -= BankAccount.OVERDRAWN_PENALTY

    @staticmethod
    def change_overdrawn_penalty(new_penalty):
        BankAccount.OVERDRAWN_PENALTY = new_penalty
